### DISCUSSION TITLE
Dialog display when starting flet

### DISCUSSION THREAD

**Discussion by gitchin** [flet-dev/flet#1745] [General]
No dialog is displayed when flet starts.

```
import flet as ft

def main(page:ft.Page):
    page.controls = [
        ft.Row(
            controls=[
                ft.Text("Test")
            ]
        )
    ]
    page.update()


    dlg = ft.AlertDialog(
        modal=True,
        content=ft.TextField(label="input")
    )

    page.show_dialog(dlg)

    page.update()


ft.app(target=main)
```

**Comment by Alan3344** [comment]
I don't know why, it was displayed for the first time, but it was not displayed for several times in a row, the chance of recurrence is too small
The code has been commented out before running

```
Python 3.11.4 (main, Jul 25 2023, 17:36:13) [Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.

pip show flet
Name: flet
Version: 0.8.4
```


![image](https://github.com/flet-dev/flet/assets/60773912/20a89099-c93d-4857-a308-7acf14d33801)
![image](https://github.com/flet-dev/flet/assets/60773912/e56761a7-9539-4ac5-a2a5-7fc19e847d8f)

**Comment by ndonkoHenri** [comment]
I happened to make it work.
Add a time delay before showing the dialog:
```python
time.sleep(0.0001)   #any value of your choice
page.show_dialog(dlg)
```
I don't know the reason for this behaviour. 😕

**Reply by FeodorFitsner** [reply]
That's approach I used myself multiple times. I don't like it honestly and I promise it's a temporary solution :)
We need to figure out why is that happening!

**Reply by gitchin** [reply]
I will use time.sleep.
Thank you.
